File: Unit2-01/java/MrCoxallStack.java
/*
 * This is a class of MrCoxallStack string stack.
 *
 * @author Ayaat Ahmedou
 * @version 1.0
 * @since 2024-10-15
 */

import java.util.ArrayList;
import java.util.List;

/**
 * This is the MrCoxallStack string stack class.
 *
 * @null This class does not handle null value.
 */
public class MrCoxallStack {
    /**
     * The constructor for the MrCoxallStack string stack class has one field.
     */
    private List<String> stackAsList;

    /**
     * This is it being assigned the ArrayList.
     */
    public MrCoxallStack() {
        this.stackAsList = new ArrayList<String>();
    }

    /**
     * Method to pusih a string into a stack.
     *
     * @param item The string that will be added.
     *
     */
    public void pushItem(String item) {
        stackAsList.add(item);
    }

    /**
     * Method to return the stack as a comma-seperated string.
     *
     * @return The string of the stack.
     *
     */
    public String showStack() {
        return String.join(", ", stackAsList);
    }
}

File: Unit2-01/ts/MrCoxallStack.ts
/**
 * MrCoxallStack string stack.
 *
 * By:    Ayaat Ahmedou
 * Version: 1.0
 * Since: 2024-10-16
 */

export class MrCoxallStack {
  // This class defines a MrCoxallStack string stack
  private readonly stackAsList: string[]

  constructor () {
    // The constructor for the MrCoxallStack string stack class.
    // Args: None
    this.stackAsList = []
  }

  // Methods

  // This method pushes a string onto the stack.
  // Args: pusedString (string): The string will be pushed onto the stack.
  pushItem (pushedString: string): void {
    this.stackAsList.push(pushedString)
  }

  // This method returns the stack as a string.
  // Returns:
  // string: The stack as  string.
  showStack (): string {
    let stackValues = ''

    // Iterating through the stack and concating the values
    for (const value of this.stackAsList) {
      stackValues += value + ', '
    }

    // Remove the last two characters (trailing comma and space)
    if (stackValues.length > 0) {
      stackValues = stackValues.slice(0, -2)
    }

    return stackValues
  }
}
